frontend/src/Initialize.tsx

```diff
@@ -216,6 +235,227 @@ const Initialize = (props: Props) => {
     );
   }
 
+  const [gameModeValue, setGameModeValue] = React.useState(modeAsString);
```

Comment:
I'm not sure why you need these `useState` values -- wouldn't it suffice to save `props.state.propagated` directly? It should be the closest match to the existing game, rather than any view-local state values

Reply:
I tried that, it did not work. I set the values in the select list first, hoping that it would trigger the change event, it did not. So I did what I did, kind of a pain.

I guess I can send the change config requests directly to the server first, then the response will cause the client to re-render its value without all these ugly codes, like resetGameSettings does! Will fix this.